### PR TITLE
Sled/instance association for inventory and insights

### PR DESCRIPTION
**Pull request by zephraph**
This PR creates a view of instances that's exposed via the sleds endpoint to provide operators an insight into what's running on what hardware. This will aid in both understanding the service impact of maintenance as well as providing debugging aids.

We've made some explicit decisions to expose the project and instance names (but as little other details as possible).

### CODE REVIEW

File: common/src/sql/dbinit.sql [hunk 1]
```diff
@@ -862,10 +863,43 @@ CREATE UNIQUE INDEX ON omicron.public.instance (
...
+JOIN
+    omicron.public.project as project
+ON  
+    instance.project_id = project.id
+JOIN
+    omicron.public.silo as silo
+ON  
+    project.silo_id = silo.id
+WHERE
+    instance.active_sled_id IS NOT NULL
```
Comment (luqmana) [inline common/src/sql/dbinit.sql:899]: I'm a little inclined to say just remove this condition. I don't think this is ever `NULL` today once the instance has been created but there's an issue (#2315) to actually make it `Nullable/Option`. But even at then, `DataStore::sled_instance_list` already adds a filter `active_sled_id` matches the given `authz_sled`.

File: nexus/db-queries/src/db/lookup.rs [hunk 2]
```diff
@@ -760,6 +760,15 @@ lookup_resource! {
     primary_key_columns = [ { column_name = "id", rust_type = Uuid } ]
 }
 
+lookup_resource! {
+    name = "SledInstance",
+    ancestors = [],
+    children = [],
+    lookup_by_name = true,
```
Comment (luqmana) [inline nexus/db-queries/src/db/lookup.rs:767]: Don't think I saw anywhere actually doing a lookup by name?

File: common/src/sql/dbinit.sql [hunk 3]
```diff
@@ -862,10 +863,43 @@ CREATE UNIQUE INDEX ON omicron.public.instance (
...
+AND instance.time_deleted IS NULL;
```
Comment (david-crespo) [inline common/src/sql/dbinit.sql:900]: Nitpick but everything after `FROM` is hard to read with all the newlines. I ran it through [`sqlfmt`](https://sqlfum.pt/) (which is by a former CRDB employee) and it came out with this:

```sql
FROM
    omicron.public.instance AS instance
    JOIN omicron.public.project AS project ON
            instance.project_id = project.id
    JOIN omicron.public.silo AS silo ON
            project.silo_id = silo.id
WHERE
    instance.active_sled_id IS NOT NULL
    AND instance.time_deleted IS NULL;
```

Do with that what you will.

File: nexus/db-queries/src/db/lookup.rs (hunk 2 above)
Comment (zephraph) [reply nexus/db-queries/src/db/lookup.rs:767]: We're not, that's true! Good catch.

**Review by david-crespo — APPROVED** [review]
Looks great! You made it look easy and obvious, which I know it wasn't.

**Review by luqmana — APPROVED** [review]

**Pull request merged**